api/src/schema/core/Collection/CollectionSchema.ts

```diff
@@ -252,8 +252,8 @@ const collection_def_input_to_model_data_fields = async (
 export const CollectionSchema = makeExecutableSchema({
   typeDefs: `
   type Query { 
-    collection(collection_id: String!): Collection
-    column_def(column_id: String!): ColumnDef
+    collection(collection_id: String!): Collection!
+    column_def(column_id: String!): ColumnDef!
```

Comment:
I double checked the resolver return types for the fields you marked non-nullable, and I believe none of them actually return `undefined` or `null` in practice (throwing our bilingual error messages in cases where they would), but the typing of the `column_def` query resolver _thinks_ it might be able to return `undefiend`. That's not true, but the TS interpreter just isn't able to figure that out on its own. 

It's not strictly speaking an issue, right now, but we should really enforce that a field is non-nullable if and only if it's resolver has a non-null and non-undefined return type. If/when we refactor away from string-based schemas, the tooling will enforce this. 

Here's one way to refactor the `column_def` query resolver to keep the same behaviour it currently has while fixing the return type:
```
      column_def: resolver_with_authz(
        async (
          _parent: unknown,
          { column_id }: { column_id: string },
          context,
          { fieldName }: { fieldName: string },
        ) => {
          const column_def = await ColumnDefByIdLoader.load(column_id);

          const operation_name = `Query \`${fieldName}\``;
          if (typeof column_def === 'undefined') {
            throw make_authroization_error(operation_name);
          } else {
            const collection = await CurrentCollectionByRecordsetKeyLoader.load(
              column_def.meta.recordset_key,
            );

            validate_collection_level_authorization(
              operation_name,
              context.req.user,
              collection,
              user_can_view_collection,
            );

            return column_def;
          }
        },
      ),
      ```
      
Note: we want it to throw the same authorization error whether the column id doesn't exist OR the query maker is lacking authorization. We can't validate authorization if the column doesn't exist, therefore letting the query-maker know whether or not the column exists is potentially an information leak.

Reply:
The return types on the rest of the resolvers you marked as non-nullable check out. Just to confirm though, have you checked that all the other fields that are still nullable in the schema _are_ nullable? It'd be good at this point (and going forward, my bad for slacking on this) to make sure the schema is accurate to the resolvers (or mongoose models, for scalar fields that don't have resolvers).

Reply:
I double checked the resolver methods ONLY (my bad for slacking off on this) for the types that I had changed based on their usage in the code. Since we are committing the generated types, I think I can handle that as part of this PR before merge.

Reply:
Modified said resolvers.